ext/liquid_c/block.c

```diff
@@ -290,32 +318,87 @@ static VALUE internal_block_body_parse(block_body_t *body, parse_context_t *pars
     return unknown_tag;
 }
 
-static void ensure_intermediate(block_body_t *body)
+typedef struct block_body_yield_tag_args {
+    block_body_t *body;
+    serialize_parse_context_t *serialize_context;
+    tag_markup_header_t *current_tag;
+} block_body_yield_tag_args_t;
+
+static VALUE block_body_try_yield_tag(VALUE uncast_args)
```

Comment:
Still make sense to have `try` in the name since the `raise` was removed?

Reply:
I think it does make sense to have `try` in the name since it's called in the context of a `try/rescue` [here](https://github.com/Shopify/liquid-c/blob/ed940df097a2a8d55c7915c726a3242496267c20/ext/liquid_c/block.c#L372).